Add a note about gitconfig username and email automatic update (#2692)
codenvy/che-docs: modules/end-user-guide/pages/mounting-git-configuration.adoc

EOF
----

> [NOTE]

Reviewer: Let's put the note at the top of the procedure, before pre-requirements

Author: I've changed the configmap block to not to specify just username and email but generalise the gitconfig content:

![screenshot-0 0 0 0_4000-2024 02 22-16_31_27](https://github.com/eclipse-che/che-docs/assets/7668752/5ac6706d-8b48-4c6a-b6b0-929df8ec15c7)

 With this change I think the note suits here more as it specifies that the general gitconfig content can be updated automatically.

Reviewer: @vinokurig this NOTE should be placed at the very beginning, just after the title `Mounting Git configuration`

Author: @ibuziuk done

====

> The `user.name` and `user.email` fields can be set automatically from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token]. The related data must be set in the provider's user profile page.

Reviewer: is it supported for all providers?

Author: yes

Reviewer: it is not only about access token, but also Git OAuth

Author: It is not related directly to Git OAuth, the idea is to have a token secret either generated by Git OAuth, or manually created PAT secret. So the token documentation reference covers both cases.

Reviewer: well, but it is linked with https://eclipse.dev/che/docs/stable/end-user-guide/using-a-git-provider-access-token/ that talks about PAT which is misleading

Author: Mentioned the OAuth source as well.

> The `user.name` and `user.email` fields can be set automatically to the `gitconfig` content from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token] or a token generated via OAuth. The related data must be set in the provider's user profile page.

Reviewer: ```suggestion
The `user.name` and `user.email` fields will be set automatically to the `gitconfig` content from a git provider, connected to {prod-short} by a xref:using-a-git-provider-access-token.adoc[Git-provider access token] or a token generated via OAuth, if username and email are set on the provider's user profile page.
```
Please, move this note above before .Prerequisites
